### PR TITLE
Convert ParameterizedBenchmark to tf.test.Benchmark.

### PR DESCRIPTION
**Pull request by xingyu-long**
~~Add PerfZeroBenchmark into benchmarks folder and use PerfZeroBenchmark as base class. It can be run both on Bazel tests and PerfZero script.~~
We decide to use tf.test.Benchmark for our benchmark.

### CODE REVIEW

File: tensorflow/python/keras/benchmarks/BUILD [hunk 1]
```diff
@@ -91,3 +92,16 @@ py_library(
...
+
+py_library(
+    name = "perfzero_benchmark_lib",
+    srcs = [
+        "perfzero_benchmark.py",
+    ],
+    deps = [
+        "@absl_py//absl/flags",
+        "@absl_py//absl/logging",
+        "@absl_py//absl/testing:flagsaver",
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/BUILD:104]: `@` looks not right.

File: tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py [hunk 2]
```diff
@@ -40,38 +39,28 @@ class KerasExamplesBenchmark(
...
-            x)
+        tf.keras.layers.LSTM(
+            64, return_sequences=True))(x)
     x = tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(64))(x)
     outputs = tf.keras.layers.Dense(1, activation='sigmoid')(x)
     model = tf.keras.Model(inputs, outputs)
     return model
 
-  def benchmark_bidirect_lstm_imdb(self, batch_size, run_iters):
+  def benchmark_bidirect_lstm_imdb(self):
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py:55]: All parameterized benchmarks should be converted to `benchmark_xxx` test. So you should have 5 benchmark methods here.

File: tensorflow/python/keras/benchmarks/perfzero_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
```
Comment (yhliang2018) [inline tensorflow/python/keras/benchmarks/perfzero_benchmark.py:1]: Copying is not good, but let's hold it here for now.

File: tensorflow/python/keras/benchmarks/BUILD (hunk 1 above)
Comment (xingyu-long) [reply tensorflow/python/keras/benchmarks/BUILD:104]: > `@` looks not right.

I checked with current tensorflow repo and use same format. You can find from [here](https://github.com/tensorflow/tensorflow/search?q=%40absl_py%2F%2Fabsl%2Flogging&unscoped_q=%40absl_py%2F%2Fabsl%2Flogging).

File: tensorflow/python/keras/benchmarks/perfzero_benchmark.py (hunk 3 above)
Comment (xingyu-long) [reply tensorflow/python/keras/benchmarks/perfzero_benchmark.py:1]: Sounds great to me.

File: tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py (hunk 2 above)
Comment (xingyu-long) [reply tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py:55]: > All parameterized benchmarks should be converted to `benchmark_xxx` test. So you should have 5 benchmark methods here.

Do you mean I need to add 5 benchmarks for different models from keras.io after this PR?

File: tensorflow/python/keras/benchmarks/BUILD (hunk 1 above)
Comment (yhliang2018) [reply tensorflow/python/keras/benchmarks/BUILD:105]: emm... that's strange. Can it work correctly if you run the benchmark locally with bazel?

File: tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py (hunk 2 above)
Comment (yhliang2018) [reply tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py:55]: In theory, for equivalent conversion, it might be good to have 5 benchmark test.

I think it also depends on how large the dataset is in the benchmark. If the dataset size is large, you may use larger batch size for it to make the benchmark run faster. So we need to know how many examples in the dataset before determining the `batch_size` number.

For this one, 1 benchmark test is not enough. If 5 is too many, let's keep it as 3?

File: tensorflow/python/keras/benchmarks/BUILD (hunk 1 above)
Comment (xingyu-long) [reply tensorflow/python/keras/benchmarks/BUILD:105]: > emm... that's strange. Can it work correctly if you run the benchmark locally with bazel?

it worked. For the current failed checks, it doesn't relate to our code from log.

File: tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py (hunk 2 above)
Comment (xingyu-long) [reply tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py:55]: So, we can keep the similar configuration like `(batch_size = 32, run_iters=2)`, `(batch_size =64, run_iters=3)`, `(batch_size =64, run_iters=1)`, is it correct? If so, I'd like rename as this format `benchmark_model_bs_xx_iter_xx`. It's more straightforward to us.

File: tensorflow/python/keras/benchmarks/BUILD (hunk 1 above)
Comment (yhliang2018) [reply tensorflow/python/keras/benchmarks/BUILD:105]: All right. It's fine as long as it works. copybara should be able to handle it then.

File: tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py (hunk 2 above)
Comment (yhliang2018) [reply tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py:55]: Maybe `benchmark_model_bs_xx`? `iter` can be ignored in the name, as it's more about how to collect benchmark results, not how the models are tested. 

Later if dist-strat is applied, # of gpus can be added to the name.

Comment (xingyu-long) [reply tensorflow/python/keras/benchmarks/keras_examples_benchmark_test.py:55]: Sounds great to me.  I will update this PR tomorrow.

**Comment by gbaned** [comment]
@xingyu-long Can you please resolve conflicts? Thanks!

**Comment by xingyu-long** [comment]
> @xingyu-long Can you please resolve conflicts? Thanks!

Since host and I decided to hold this PR for now, but I will talk to her about this.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by xingyu-long** [comment]
Already update in another [PR](https://github.com/tensorflow/tensorflow/pull/41456)
